[ImportVerilog] `inside` expression only support simple bit vectors

The BlackParrot core in the sv-tests repository is currently failing on the `inside` expression only supporting simple bit vectors. I suspect that this is due to some enum/struct/union being passed to `inside`, which should be converted to a simple bit vector:

https://chipsalliance.github.io/sv-tests-results/?v=circt_verilog+black-parrot+bp_default

```
third_party/cores/black-parrot/bp_me/src/v/cce/bp_uce.sv:290:43: error: only simple bit vectors supported in 'inside' expressions
  wire miss_load_v_li   = cache_req_v_i & cache_req_cast_i.msg_type inside {e_miss_load};
```

--- a/lib/Conversion/ImportVerilog/Expressions.cpp
+++ b/lib/Conversion/ImportVerilog/Expressions.cpp
@@ -637,12 +637,7 @@ struct RvalueExprVisitor {
         cond = builder.create<moore::AndOp>(loc, leftValue, rightValue);
       } else {
         // Handle expressions.
-        auto ty = listExpr->type;
-        bool isUnPackedStruct = ty->isUnpackedStruct();
-        bool isUnPackedArray = ty->isUnpackedArray();
-        bool isUnPackedUnion = ty->isUnpackedUnion();
-
-        if (isUnPackedArray || isUnPackedStruct || isUnPackedUnion) {
+        if (!listExpr->type->isIntegral()) {
           if (listExpr->type->isUnpackedArray()) {
             mlir::emitError(
                 loc, "unpacked arrays in 'inside' expressions not supported");

--- a/lib/Conversion/ImportVerilog/Expressions.cpp
+++ b/lib/Conversion/ImportVerilog/Expressions.cpp
@@ -637,14 +637,9 @@ struct RvalueExprVisitor {
         cond = builder.create<moore::AndOp>(loc, leftValue, rightValue);
       } else {
         // Handle expressions.
-        if (!listExpr->type->isSimpleBitVector()) {
-          if (listExpr->type->isUnpackedArray()) {
-            mlir::emitError(
-                loc, "unpacked arrays in 'inside' expressions not supported");
-            return {};
-          }
+        if (listExpr->type->isUnpackedArray()) {
           mlir::emitError(
-              loc, "only simple bit vectors supported in 'inside' expressions");
+              loc, "unpacked arrays in 'inside' expressions not supported");
           return {};
         }
         auto value = context.convertToSimpleBitVector(

--- a/lib/Conversion/ImportVerilog/Expressions.cpp
+++ b/lib/Conversion/ImportVerilog/Expressions.cpp
@@ -637,11 +637,22 @@ struct RvalueExprVisitor {
         cond = builder.create<moore::AndOp>(loc, leftValue, rightValue);
       } else {
         // Handle expressions.
-        if (listExpr->type->isUnpackedArray()) {
+        auto ty = listExpr->type;
+        bool isUnPackedStruct = ty->isUnpackedStruct();
+        bool isUnPackedArray = ty->isUnpackedArray();
+        bool isUnPackedUnion = ty->isUnpackedUnion();
+
+        if (isUnPackedArray || isUnPackedStruct || isUnPackedUnion) {
+          if (listExpr->type->isUnpackedArray()) {
+            mlir::emitError(
+                loc, "unpacked arrays in 'inside' expressions not supported");
+            return {};
+          }
           mlir::emitError(
-              loc, "unpacked arrays in 'inside' expressions not supported");
+              loc, "only simple bit vectors supported in 'inside' expressions");
           return {};
         }
+
         auto value = context.convertToSimpleBitVector(
             context.convertRvalueExpression(*listExpr));
         if (!value)

--- a/test/Conversion/ImportVerilog/basic.sv
+++ b/test/Conversion/ImportVerilog/basic.sv
@@ -1545,12 +1545,8 @@ module Expressions;
     // CHECK: moore.struct_create [[TMP0]], [[TMP1]] : !moore.i32, !moore.i32 -> struct<{a: i32, b: i32}>
     struct0 = '{43, 9002};
 
-    // CHECK: [[TMP0:%.+]] = moore.constant 2
-    // CHECK: [[TMP1:%.+]] = moore.constant 3
-    // CHECK: moore.struct_create [[TMP0]], [[TMP1]] : !moore.i32, !moore.i32 -> struct<{a: i32, b: i32}>
-    struct0 = '{2,3};
 
-    // CHECK: [[TMP0:%.+]] = moore.constant 2 : i32
+    // CHECK: [[TMP0:%.+]] = moore.constant 43 : i32
     // CHECK: [[TMP1:%.+]] = moore.sext [[TMP0]] : i32 -> i64
     // CHECK: [[TMP2:%.+]] = moore.read %struct0 : <struct<{a: i32, b: i32}>>
     // CHECK: [[TMP3:%.+]] = moore.conversion [[TMP2]] : !moore.struct<{a: i32, b: i32}> -> !moore.i64
@@ -1559,7 +1555,7 @@ module Expressions;
     // CHECK: [[TMP6:%.+]] = moore.conversion [[TMP5]] : !moore.i32 -> !moore.l32
     // CHECK: [[TMP7:%.+]] = moore.conversion [[TMP6]] : !moore.l32 -> !moore.i32
     // CHECK: moore.blocking_assign %c, [[TMP7]] : i32
-    c = 2 inside {struct0};
+    c = 43 inside {struct0};
 
     // CHECK: [[TMP0:%.+]] = moore.constant 44
     // CHECK: [[TMP1:%.+]] = moore.constant 9003

--- a/test/Conversion/ImportVerilog/basic.sv
+++ b/test/Conversion/ImportVerilog/basic.sv
@@ -1545,6 +1545,22 @@ module Expressions;
     // CHECK: moore.struct_create [[TMP0]], [[TMP1]] : !moore.i32, !moore.i32 -> struct<{a: i32, b: i32}>
     struct0 = '{43, 9002};
 
+    // CHECK: [[TMP0:%.+]] = moore.constant 2
+    // CHECK: [[TMP1:%.+]] = moore.constant 3
+    // CHECK: moore.struct_create [[TMP0]], [[TMP1]] : !moore.i32, !moore.i32 -> struct<{a: i32, b: i32}>
+    struct0 = '{2,3};
+
+    // CHECK: [[TMP0:%.+]] = moore.constant 2 : i32
+    // CHECK: [[TMP1:%.+]] = moore.sext [[TMP0]] : i32 -> i64
+    // CHECK: [[TMP2:%.+]] = moore.read %struct0 : <struct<{a: i32, b: i32}>>
+    // CHECK: [[TMP3:%.+]] = moore.conversion [[TMP2]] : !moore.struct<{a: i32, b: i32}> -> !moore.i64
+    // CHECK: [[TMP4:%.+]] = moore.wildcard_eq [[TMP1]], [[TMP3]] : i64 -> i1
+    // CHECK: [[TMP5:%.+]] = moore.zext [[TMP4]] : i1 -> i32
+    // CHECK: [[TMP6:%.+]] = moore.conversion [[TMP5]] : !moore.i32 -> !moore.l32
+    // CHECK: [[TMP7:%.+]] = moore.conversion [[TMP6]] : !moore.l32 -> !moore.i32
+    // CHECK: moore.blocking_assign %c, [[TMP7]] : i32
+    c = 2 inside {struct0};
+
     // CHECK: [[TMP0:%.+]] = moore.constant 44
     // CHECK: [[TMP1:%.+]] = moore.constant 9003
     // CHECK: moore.struct_create [[TMP0]], [[TMP1]] : !moore.i32, !moore.i32 -> ustruct<{a: i32, b: i32}>

